packages/fillings/src/createRuntimeFillings.ts

```diff
@@ -0,0 +1,123 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function createRuntimeFillings(config: any): any {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (options: any) => {
+    if (typeof options !== 'object' || options === null) {
+      return {
+        className: '',
+        assignVars() {
+          return {};
+        },
+      };
+    }
+
+    if (config.defaultCondition) {
+      const fillings = config as ConditionalFillingsConfig<
+        BaseConditions,
+        string
+      >;
+      const props = options as ComputeFillingsProps<typeof fillings>;
+
+      const { defaultCondition, definitions, properties } = fillings;
+
+      const usedProps = Object.keys(props).filter((prop) =>
+        properties.includes(prop),
+      ) as (keyof typeof props)[];
+
+      if (usedProps.length === 0) {
+        return {
+          className: '',
+          assignVars() {
+            return {};
+          },
+        };
+      }
+
+      const classNames: string[] = [];
+      const setVars: [CSSVarFunction, string][] = [];
+
+      for (const prop of usedProps) {
+        if (typeof props[prop] === 'object' && props[prop] !== null) {
+          const valuesByCondition = props[prop] as Record<string, string>;
+
+          for (const conditionName in valuesByCondition) {
+            classNames.push(definitions[prop][conditionName].className);
+
+            setVars.push([
+              definitions[prop][conditionName].variable,
+              valuesByCondition[conditionName],
+            ]);
+          }
+        }
+
+        if (typeof props[prop] === 'string') {
+          classNames.push(definitions[prop][defaultCondition].className);
+
+          setVars.push([
+            definitions[prop][defaultCondition].variable,
+            props[prop] as string,
+          ]);
+        }
+      }
+
+      return {
+        className: classNames.join(' '),
+        assignVars() {
+          return assignInlineVars(Object.fromEntries(setVars));
+        },
+      };
+    }
+
+    const fillings = config as FillingsConfig<string>;
+    const props = options as ComputeFillingsProps<typeof fillings>;
+    const { definitions, properties } = fillings;
+
+    const usedProps = Object.keys(props).filter((prop) =>
+      properties.includes(prop),
+    ) as (keyof typeof props)[];
+
+    if (usedProps.length === 0) {
+      return {
+        className: '',
+        assignVars() {
+          return {};
+        },
+      };
+    }
+
+    const classNames: string[] = [];
+    const setVars: [CSSVarFunction, string][] = [];
+
+    for (const prop of usedProps) {
+      if (typeof props[prop] === 'string') {
+        classNames.push(definitions[prop].className);
+
+        setVars.push([definitions[prop].variable, props[prop] as string]);
+      }
+    }
+
+    return {
+      className: classNames.join(' '),
+      assignVars() {
+        return assignInlineVars(Object.fromEntries(setVars));
```

Comment:
I'd probably avoid `Object.fromEntries` as it was only supported in Safari 12+.

Reply:
If it will broaden supported browsers list for this package we'll update it :)